Functions to calculate the closest bigger city than the given one and up to the biggest in the country.

kermit10000000 opened the pull request:
The script works and you have a given example in main.cpp
Maybe you have some idea of further usage of this like for each capital choosen by the best capital scritpt or something else? :) 

Signed-off-by: Wojciech Mielczarek <kermit10000000@gmail.com>

JaroslawWiosna commented on ClosestCity/ClosestCity.cpp:
```diff
@@ -0,0 +1,70 @@
+
+namespace ClosestCity {
+
+void removeCites(std::vector<City> &vec, const std::string &cityName) {
```
pls, rename to `removeSmallerCitiesInVector` or `removeSmallerCities`

JaroslawWiosna commented on main.cpp:
```diff
@@ -59,12 +60,16 @@ int main(int argc, char *argv[]) {
 
     VectorOfCities::setDistanceToTheFarthestInTheWholeVector(vec);
 
+
```
remove this line

JaroslawWiosna commented on ClosestCity/ClosestCity.cpp:
```diff
@@ -0,0 +1,70 @@
+    std::string cities_list = cityName;
+    std::string next_city_in_line = "";
+    while (cityName != " ") {
+        removeCites(vec, cityName);
```
Please think about removing this line.
Perhaps it is not needed for performance.... But for now, you can merge it.

kermit10000000 commented:
1. fixed
2. performance to be checked in the future
3. fixed
4. empty() instead of  " " fails for some unknown reason. maybe changed in the future.

kermit10000000 commented:
I know this pull request is closed but since we discussed it here, about second point 3 options should be checked: 1. performance with no remove function, 2. performance with remove in while(!="") and 3 . only once called remove function before the while(!="")
